### ISSUE TITLE
Adjustable font scaling

### ISSUE THREAD

**Issue by edgimar** [signalapp/Signal-Android#1509]
It would be nice to be able to scale the font size of messages with respect to the system font size.  Sometimes it is convenient to see more conversation history on the screen at the expense of smaller sizes and reduced readability.

**Comment by wp9015362** [comment]
+1

Please add an option to TextSecure which allows to adjust the font size within TextSecure (i.e. independenty from system font size).

It would be much appreciated.

Regards

**Comment by wp9015362** [comment]
@mcginty :

On the mailing list I saw you post this:

> @mcginty wrote in https://lists.riseup.net/www/arc/whispersystems/2015-04/msg00010.html
> 
> Right now TextSecure should honor the system-wide relative font sizes (I 
> tried to use “sp” units whenever possible with text). I didn’t know there was 
> much demand for setting the font size specifically in one app. Is the font 
> too large or small specifically in TextSecure? I guess other apps do tend to 
> have font size options in their settings.

I refuse to answer on the mailing list because the mailing list archive exposes the email addresses of each poster, which I don't like for privacy reasons.

And I also think it's more comfortable to post on GitHub than on something as antiquated as a mailing list. And I simply don't want to subscribe to a mailing list when I already have a GitHub account.

So I am going to answer here:

Yes, there is demand for adjustable message font size (as you can see from this featuree request for example).

And yes, other apps do have font size options in their settings, like Telegram and WhatsApp for example.

By the way, speaking about font sizes, there currently is a bug in TextSecure, because the font size in the message input box is different than the font size for messages, see:

https://github.com/WhisperSystems/TextSecure/issues/2769

It would be much appreciated if you could implement a font size option into TextSecure and fix that bug.

The font size option additionally could also have a "Respect system font size" setting so that users who like the current approach could still go on using it, i.e. with font size set to "Respect system font size" (or whatever you wanna call it) the message font size would still scale with the system font size.

**Comment by moxie0** [comment]
@wp9015362 Then please don't post at all.  If you continue posting discussions from the mailing list here, we'll ban you from the repo.

**Comment by therealpxc** [comment]
One use case is for the severely visually impaired. My mother is extremely visually impaired (legally blind, uncorrectable), but she gets by with a physical magnifier, a phablet, large fonts, TTS, and dictation. Unfortunately accessibility tends to get short shrift on Android, and with recent versions she is simply unable to set fonts large enough to use without a magnifier. I would be very grateful if Signal were usable for her.

**Comment by hiredgunhouse** [comment]
+1 for what @therealpxc requests
NOTE:
#4894 that is about both making font smaller and bigger has been closed because of this one and this one has been deferred because the OP wanted to actually make fonts smaller and @mcginty made a valid point that when you want fonts smaller you probably want it for the whole system, but as @therealpxc wrote and the referenced issue states there are cases when you want fonts bigger and bigger only in Signal (like many other apps let you do).

**Comment by tserong** [comment]
The default Android messenger app overrides the volume button to increase/decrease the font size when viewing message threads. ISTM it would be ideal if the same functionality were available in Signal, via the same means. This has definite accessibility advantages, and as it would work the same was as the default messenger app, people will already be used to this behaviour.

**Comment by zuiram** [comment]
What @tserong said. I searched every option to set this up for my dad, then googled and found this, clarifying that he needs to use the default app instead.

Zoom is an accessibility feature that affects app viability also for the mildly vision impaired, like him, who get by fine normally with system scaling, but adverse reading conditions (e.g. wrong light) can make zooming necessary.

The volume rocker as a zoom input is the POLA approach, but pinch zoom is also a very familiar gesture, and even a static scale multiplier buried in the settings or convo settings would help some users, though I realize it's probably more involved than it seems to implement.

**Comment by TeddyDD** [comment]
This is pretty important for people with vision defects. Signal is unusable for some people because they are not able to make fonts bigger.

**Comment by aroxby** [comment]
I know this issue is pretty old, but it's still open so I'll add my $0.02.
In the default Android app (or at least it's the default), you can pinch to zoom in on SMS.  I find this feature very helpful.

TLDR: +1

**Comment by FeuRenard** [comment]
Fixed by 5f5db5ded7382d8f55c40c0e4df669828f8e0bdc

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
